### ISSUE TITLE
Find a VA Form: Add Relevance, Popularity, Related Forms & UI consistency

### ISSUE THREAD

**Issue by MarciMcGuireGCIO** [department-of-veterans-affairs/va.gov-team#21639]
## User Story

This Epic will encompass the following:

-  [x] Design & FE changes to reflect **new search tool UI consistency pattern**
-  [x] (Lighthouse) Forms API improvements to handle misspellings, search for root words, then score and rank results by relevance.
- [x] (Lighthouse) Forms API improvements to implement **popularity** data
-  [x] FE change to add "Best Match" to the search results sorting feature
-  ~~[ ] FE change to add "Most Popular" to the search results sorting feature~~ NOT A SORTING OPTION, will be used behind the scenes to weight results
- ~~[ ] Design & FE change to add PDF download information due to issue with some forms not being supported by browsers~~ DESCOPED 6/14/2021
-  [x] Design & FE change to add link to "Related Forms" listed on a form detail page

## Expected Behavior

Matches are found based on finding complete phrases in the form name or form numbers that are transformed in order to better match the form number or name in the VA database.   

## Acceptance Criteria

- All issues associated with this Epic are completed, and go through the recommended steps in the VSP Collab Cycle:
   - [x]  Design Intent
   - [x]  Midpoint Review
  - [x] Contact Center Review (including documentation updates)
   - [x]  Staging Review (including all a11y and QA prep/testing)
   - [x]  Privacy & Security Review
   - [ ]  Full Accessibility & 508 Office Audit
   - [ ]  Post-Launch Check-In

## Tickets to create
- Content request for "Popularity/Most Popular" and for PDF download instructions #25578
- FE task to add popularity option to the sort feature.  #25566
- New contact center review  
~~- New issue to add PDF download info to all PDF links (blocked by design being finalized) -  #25571~~

**Comment by johnhashva** [comment]
Adding here: @kelsonic's work on Resources & Support for whole word matching logic: -- https://github.com/department-of-veterans-affairs/va.gov-team/issues/16916

This is the current [logic](https://github.com/department-of-veterans-affairs/vets-website/blob/40bc21ad6e94fc15e5ae3af314fc45cd5bd90bce/src/applications/resources-and-support/hooks/useGetSearchResults.js#L53)

Non-prod envs:
Experiment with new sorting on non-prod envs.
We derive the number of times a keyword is found in the article's title.
We also derive the number of times a keyword is found in the article's description.
We sort first by query word instances found in title descending
We sort ties then by query word instances found in description descending
We sort ties then by title ascending

Good start here would be to look at FaF search queries and see what patterns emerge for whole word searching of forms then determine whether the Kelson approach here or a variation would be optimal.

**Comment by MarciMcGuireGCIO** [comment]
Perfect!  Thanks, @johnhashva !

**Comment by MarciMcGuireGCIO** [comment]
@zacharymorel we have some other tickets in the backlog that may overlap with this - particularly where Kelson is ranking based on where the search term is matched.  Even if there turns out to be no overlap, they are probably still good to keep in mind as we think of possible solutions.

- #19170
- #10754
- #20273

**Comment by johnhashva** [comment]
Adding the full fetching and ordering logic that @kelsonic has created for R&S:

FETCHING RESULTS + FILTERING: 
Lowercase each word in the search query. 
Remove ignored words from search query (e.g. "in", "the", etc.) 
Unpluralize search query words (e.g. so that a search for "disabilities" will still yield articles titled "disability") Filter out articles that do not contain any search query words within its title.  

ORDERING: 
Sort first by the number of exact query matches (ignoring casing) in the title and description 
Sort ties by query word instances found in title descending 
Sort ties then by query word instances found in description descending 
Sort ties then by alphabetical descending

**Comment by MarciMcGuireGCIO** [comment]
This ticket is covered in another issue.  Closing.

**Comment by MarciMcGuireGCIO** [comment]
Updating title to reflect total scope of this epic.

**Comment by MarciMcGuireGCIO** [comment]
Noting here that we will clarify search logic with Lighthouse team on 6/7 - need to determine if sorting can be for Best Match or Popularity or if they are one in the same.

**Comment by MarciMcGuireGCIO** [comment]
Updating to note that they relevance (Best Match) and popularity are combined.  Popularity data is largely being used as a tiebreaker and to weight results.
